src/web/components/MainMedia.tsx

```diff
@@ -45,18 +48,36 @@ function renderElement(
     pillar: Pillar,
     i: number,
     hideCaption?: boolean,
+    adTargeting?: AdTargeting,
 ) {
     switch (element._type) {
         case 'model.dotcomrendering.pageElements.ImageBlockElement':
             return (
-                <MainImageComponent
+                <ImageComponent
+                    key={i}
+                    element={element}
+                    pillar={pillar}
+                    hideCaption={hideCaption}
+                    role={element.role}
+                />
+            );
+        case 'model.dotcomrendering.pageElements.YoutubeBlockElement':
+            return (
+                <YouTubeComponent
                     key={i}
                     element={element}
                     pillar={pillar}
                     hideCaption={hideCaption}
+                    // tslint:disable-next-line react-a11y-role
+                    role="inline"
```

Comment:
So we definitely seem to have a 'role' / 'weighting' name problem. I've been leaning in on role but looking at the codebase whilst that does seem to be the dominant variant the same string is also used in other contexts which isn't ideal.

Weighting may well be the way to go but this will need more investigation. I'm pretty sure role isn't confined to DCR and exists in other places so we might need a strategy in place to prevent errors when changing things. I do think it's worth knocking stuff like this on the head early though.

Reply:
So.. Ok, maybe we need to define the 'source of truth'. I believe that to be CAPI, which uses `role` so even though that diverges from Composer, it might be worth keeping as is along that principle?

Reply:
There are more inconsistencies between flixible-content (Composer)/media-api (Grid), CAPI and then frontends. Sometimes distinct fields are actually called the same (`Credit` in Grid means `Source` in CAPI; `Credit` in CAPI means Grid’s `Byline`+`Credit`!).

Gareth’s suggestion is, as always, sensible, I think.

Reply:
Plus, there are cultural habits: here, unfortunatley, everyone says `weighting` 😞 .

Reply:
👍 Lets add a comment above [`RoleType` definition](https://github.com/guardian/dotcom-rendering/blob/0832b716b8ca59cfaa982224a04e6fd0bbb5bf29/src/lib/content.d.ts#L30) explaining 'relationship' (synonymous) with weighting and why we chose role!

Reply:
👍 I added a comment